nixos/modules/services/mail/maddy.nix

```diff
@@ -391,6 +411,8 @@ in {
           ) cfg.tls.certificates)} ${optionalString (cfg.tls.extraConfig != "") ''
             { ${cfg.tls.extraConfig} }
           ''}
+        '' else if (cfg.tls.loader == "acme") then ''
+          tls acme { ${cfg.tls.extraConfig} }
```

Comment:
Having some sort of code generation that does more than just putting the extraConfig in would be nice. For example challenge, agreed, store path, hostname would be examples of options, I'd like to see available via nix options.

And we _need_ to address secrets handling. Just putting the DNS API Keys into the config isn't a good idea. With maddy, it's possibly to load variables from environment variables: https://maddy.email/reference/config-syntax/#environment-variables
Because the various DNS providers require different secrets (most a api_key, some username + password and so on), we could maybe have something around the lines of `maddy.secrets.MY_SECRET_NAME = "/run/mysecret";`, load them to environment variables in the startup script and add a note for users on the `{env:MY_SECRET_NAME}` syntax in the description.